calendra/oceania/new_zealand.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+from datetime import date, timedelta
+
+from ..core import WesternCalendar, ChristianMixin
+from ..core import MON, SAT, SUN
+from ..registry_tools import iso_register
+
+
+@iso_register("NZ")
+class NewZealand(WesternCalendar, ChristianMixin):
+    "New Zealand"
+    include_good_friday = True
+    include_easter_monday = True
+    include_boxing_day = True
+
+    FIXED_HOLIDAYS = WesternCalendar.FIXED_HOLIDAYS + (
+        (1, 2, "Day after New Year's Day"),
+        (2, 6, "Waitangi Day"),
+        (4, 25, "ANZAC Day")
+    )
+
+    def get_queens_birthday(self, year):
+        return (
+            NewZealand.get_nth_weekday_in_month(year, 6, MON, 1),
+            "Queen's Birthday"
+        )
+
+    def get_labour_day(self, year):
+        return (
+            NewZealand.get_nth_weekday_in_month(year, 10, MON, 4),
+            "Labour Day"
+        )
+
+    def get_variable_days(self, year):
+        # usual variable days
+        days = super(NewZealand, self).get_variable_days(year)
+        days.append(self.get_queens_birthday(year))
+        days.append(self.get_labour_day(year))
+
+        waitangi_day = date(year, 2, 6)
+        if waitangi_day.weekday() in self.get_weekend_days():
+            days.append((
+                self.find_following_working_day(waitangi_day),
+                "Waitangi Day Shift")
+            )
+
+        anzac_day = date(year, 4, 25)
+        if anzac_day.weekday() in self.get_weekend_days():
+            days.append((
+                self.find_following_working_day(anzac_day),
+                "ANZAC Day Shift")
+            )
+
+        christmas = date(year, 12, 25)
+        boxing_day = date(year, 12, 26)
+        if christmas.weekday() is SAT:
+            shift = self.find_following_working_day(christmas)
+            days.append((shift, "Christmas Shift"))
```

Comment:
Avoid adding "Shift" holidays. Instead, add Holiday classes that designate how they shift (default may be "following working day").

Reply:
As above.